src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java

```diff
@@ -73,6 +73,7 @@ public static void registerProvider(final IStandardRequestManager manager, final
         resolverListBuilder.addAll(ResolverHandler.registerResolvers(manager, provider.getResolvers()));
 
         manager.getProviderResolverMap().put(provider.getToken(), resolverListBuilder.build());
+        manager.getColony().markDirty();
     }
 
     public static void removeProvider(final IStandardRequestManager manager, final IToken token) throws IllegalArgumentException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)